plugins/module_utils/network/sonic/config/ospfv2/ospfv2.py

```diff
@@ -0,0 +1,765 @@
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.code)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_ospfv2_facts = self.get_ospfv2_facts()
+
+        result['before'] = existing_ospfv2_facts
+        if result['changed']:
+            result['after'] = changed_ospfv2_facts
+
+        new_config = changed_ospfv2_facts
+        old_config = existing_ospfv2_facts
+        if self._module.check_mode:
+            result.pop('after', None)
+            existing_ospfv2_facts = remove_empties_from_list(existing_ospfv2_facts)
+            is_overridden = False
+            for cmd in commands:
+                if cmd['state'] == 'overridden':
+                    is_overridden = True
+                    break
+            if is_overridden:
+                new_config = get_new_config(commands, existing_ospfv2_facts, TEST_KEYS_overridden_diff)
+            else:
+                new_config = get_new_config(commands, existing_ospfv2_facts, TEST_KEYS_diff)
+            new_config = remove_empties_from_list(new_config)
+            result['after(generated)'] = self._post_process_generated_output(new_config)
+            self.sort_lists_in_config(result['after(generated)'])
+
+        if self._module._diff:
+            result['diff'] = get_formatted_config_diff(old_config, new_config, self._module._verbosity)
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_ospfv2_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = self._module.params['config']
+        have = existing_ospfv2_facts
+        want = remove_empties_from_list(want)
+        new_want, new_have = self.validate_and_normalize_config(want, have)
+        self.sort_lists_in_config(new_want)
+        self.sort_lists_in_config(new_have)
+        resp = self.set_state(new_want, new_have)
+        return to_list(resp)
+
+    def validate_and_normalize_config(self, want, have):
+        """ Validate the configuration
+        """
+        new_want = deepcopy(want)
+        new_have = deepcopy(have)
+        for cfg in new_have:
+            self._normalize_passive_intf(cfg)
+        if new_want:
+            for cmd in new_want:
+                have_cmd = next((cfg for cfg in new_have if cfg['vrf_name'] == cmd['vrf_name']), None)
+                state = self._module.params['state']
+                self._normalize_passive_intf(cmd)
+                if state == 'merged':
+                    if have_cmd:
```

Comment:
When state is 'merged' and have_cmd is None, please validate 'default_passive' and 'passive_interfaces' config.

Reply:
Updated.